### ISSUE TITLE
CSS file returning HTTP 500 (internal server error) when pushed to BigCommerce store

### ISSUE THREAD

**Issue by jbruni** [bigcommerce/cornerstone#2155]
### Expected behavior

It should just load the theme's CSS file, as usual.

### Actual behavior

BigCommerce server returns an internal server error (HTTP code 500).

![image](https://user-images.githubusercontent.com/69181/147009447-48bac58d-cf7d-4409-8b84-803632bb7b28.png)

### Steps to reproduce behavior

Super simple: just add the following style in a theme's SCSS file, and push the theme to a BigCommerce store.

```scss
.anything {
    gap: min(13%, 57px);
}
```

This snippet of CSS passes all SCSS linting, and actually works in browsers. It should not break the entire theme's CSS. Or, at least, a hint about the issue should be provided. It has been like finding a needle in the haystack to find out this was the exact CSS rule breaking the server, since it works fine in local development with `stencil start`.

Thank you!

**Comment by jbruni** [comment]
Just found another simple/easy way to reproduce it. In fact, another valid CSS which BigCommerce currently can't handle and is breaking without letting us know what happened (the error message is only "_an error occurred_" and we feel helpless):

```scss
.anything {
    box-shadow: 0 3px 6px #00000029;
}
```

It looks like BC does not like this RGBA color notation.

I hope you BC guys can have this fixed as soon as possible - at least providing a usable error message.

Thank you!

**Comment by nickjackson** [comment]
@jbruni do you have any more info on this? We think we may be hitting a similar issue.

**Comment by BC-krasnoshapka** [comment]
hi,
most likely this is not related to Cornerstone as CSS is assembled by another service, not theme itself. I suggest you contact customer support regarding this case.

**Comment by jbruni** [comment]
Hi @nickjackson and @BC-krasnoshapka 

Yes, I have contacted BigCommerce support through the partner portal (I work for a BC partner agency), and this is the most recent answer I got:

Subject: Inconsistent SCSS linting leading to server errors

**BigCommerce Response**:

Hey again João,

I hope you had a wonderful Christmas Holiday! 

Thank you for confirming my understanding of the issue here. I was able to confirm that this is indeed an issue with our storefront theme engine, so I've gone ahead and created an internal ticket to bring this behavior to the attention of the appropriate engineering team. It seems that ultimately the issue here is that our theme engine doesn't seem to provide support for the latest version of CSS. I've gone ahead and linked this support case to the engineering ticket, as this will ensure that you're sent an email when a fix is being rolled out on the platform. You can reference the ticket by STRF-9585.

In the meantime, it seems that the only real workaround here is to avoid using the newer CSS syntax that is causing the issues. 
 
As per our support process, I will go ahead and close out this case since I've linked it to the engineering ticket. Please understand that closing this case does not mean the issue is no longer being worked on; the engineering team will continue working towards a solution and you will receive an email when the fix is being rolled out on the platform. If you have any additional questions or concerns, you can reply directly to this email and your case will be re-opened.  

Thank you for being a part of BigCommerce. Have a great rest of your day, and a Happy New Year! 

Regards,
Jordan
Product Support Engineer

**Comment by BC-krasnoshapka** [comment]
Yes, I suggest to close this one since it's not Cornerstone issue.

**Comment by nickjackson** [comment]
@jbruni @BC-krasnoshapka thanks for the heads-up. BC caught us out there. We would have expected an error while developing locally, or pushing the theme when the SCSS is bad.

**Comment by jp06** [comment]
We got bitten by this too because of color hex with alpha. Refactored it to use `rgba()` to make the status 500 go away.

If you are using the Stylelint npm script in the pipeline, I guess you can just add the [`color-hex-alpha`](https://stylelint.io/user-guide/rules/color-hex-alpha/) rule to `.stylelintrc` for now to stop that kind of code from passing through, but you also have to bump the Stylelint-related dependencies of the theme to current latest to be able to `npm run stylelint`. So far at least, it seems to work fine.

**Comment by eichenlauba** [comment]
Hi! 
Same problem here.. i've edit the cornerstone theme for my client, the stylelint doesn't return errors, same as the stencil bundle or stencil push but my front online return a 500 for the main css file.. in localhost everything work like a charm.

How can in found the css rule who crashed my compilation?

Thanks for your help
